### PR TITLE
change how osgviz singleton is created to fix segfault on close

### PR DESCRIPTION
**Pull request by arneboe**
Parts of the osg scene graph depend on the osgText::DefaultFont singleton to exist upon destruction.
Depending on the order of library loads sometimes the osg singleton was destroyed before osgviz thus resulting in a segfault when destroying osgviz.

This patch ensures that osgviz is always destroyed first.

### CODE REVIEW

**Comment by arneboe** [comment]
@planthaber Das fixed endlich den segfault beim beenden von osgviz :)

File: src/OsgViz.cpp [hunk 1]
```diff
@@ -16,19 +16,24 @@
...
+    //     If instance is global the static initialization order depends on the order in which 
+    //     libraries are loaded which leads to strange crashes in combination with osg's own singletons
+    static osg::ref_ptr<OsgViz> instance = new OsgViz(argc, argv);
+    return instance;
 }
 
 
 osg::ref_ptr<OsgViz> OsgViz::getExistingInstance(){
-	return instance;
+	return getInstance();
```
Comment (planthaber) [inline src/OsgViz.cpp:36]: This makes it impossible for Modules to check if an instance exists or not.

File: src/OsgViz.cpp [hunk 2]
```diff
@@ -16,19 +16,24 @@
...
-	return instance;
+    //HACK to work around a static destruction order bug: default font needs to be initialized before
+    //     OsgViz. Calling getDefaultFont() here ensures that the font is instanciated before 
+    //     OsgViz and thus lives longer than OsgViz and can still be called during OsgViz destruction.
+    osgText::Font::getDefaultFont();
+    //NOTE it is important that instance is a local static variable because we need a way to 
+    //     control static initialization/destruction order.
+    //     If instance is global the static initialization order depends on the order in which 
+    //     libraries are loaded which leads to strange crashes in combination with osg's own singletons
+    static osg::ref_ptr<OsgViz> instance = new OsgViz(argc, argv);
```
Comment (planthaber) [inline src/OsgViz.cpp:30]: This creating a new object each time, this changes a Method which returns a singleton into one that creates a new one on each call.
Modules that should share the same instance (and windows) cannot obtain the same instance of OsgViz anymore.

Comment (arneboe) [reply src/OsgViz.cpp:30]: No it does not ;) 
the instance is local static. the new will only be executed once.
http://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

File: src/OsgViz.cpp (hunk 1 above)
Comment (arneboe) [reply src/OsgViz.cpp:36]: I don't see any reason why a module should be able to check if an instance exists.
OsgViz has singleton semantics, i.e. it always exists.

File: src/OsgViz.cpp (hunk 2 above)
Comment (arneboe) [reply src/OsgViz.cpp:30]: afaik this is the only way to control at what point in the execution order a singleton is created. That control is mandatory if we want to use OsgViz in a multi-library environment such as rock where other singletons might exist and singletons depend on each other.

In a good design singletons should never depend on each other but that is a design flaw in osg that we cannot fix :)

File: src/OsgViz.cpp (hunk 1 above)
Comment (arneboe) [reply src/OsgViz.cpp:36]: Is there any code that checks the returned pointer before usage? I couldn't find any.

**Pull request merged**